server/src/main/java/org/elasticsearch/index/mapper/DataStreamTimestampFieldMapper.java

```diff
@@ -165,10 +165,11 @@ public void doValidate(MappingLookup lookup) {
             Map<?, ?> configuredSettings = XContentHelper.convertToMap(BytesReference.bytes(builder), false, XContentType.JSON).v2();
             configuredSettings = (Map<?, ?>) configuredSettings.values().iterator().next();
 
-            // Only type, meta and format attributes are allowed:
+            // Only type, meta, and format, and locale attributes are allowed:
             configuredSettings.remove("type");
             configuredSettings.remove("meta");
             configuredSettings.remove("format");
+            configuredSettings.remove("locale");
```

Comment:
I'm going to cover https://github.com/elastic/elasticsearch/pull/112796, but I don't know more than anybody else on this class and its implications -- if this is time sensitive, then its best that somebody else cover it rather than me.

Reply:
If the format is configurable, it makes sense that the locale is also configurable (as that affects the formats) - but I don't know what consequences arise from this

Reply:
@martijnvg do you know the answer to this question? Are you ok with this change?

Reply:
Yes, this ok. 

Other mapping attributes may affect the usability of the `@timestamp` field, which could break the promise of data streams of having a usable `@timestamp` field. The `locale` attribute and others mentioned here don't affect this, so it ok to allow these to be modified.